controllers/applications.ts

```diff
@@ -0,0 +1,147 @@
+import { addLog } from "../models/logs";
+const { logType } = require("../constants/logs");
+import { CustomRequest, CustomResponse } from "../types/global";
+const { INTERNAL_SERVER_ERROR } = require("../constants/errorMessages");
+const ApplicationModel = require("../models/applications");
+const { API_RESPONSE_MESSAGES } = require("../constants/application");
+const { getUserApplicationObject } = require("../utils/application");
+const admin = require("firebase-admin");
+
+const getAllOrUserApplication = async (req: CustomRequest, res: CustomResponse): Promise<any> => {
+  try {
+    const { userId, status, next, size } = req.query;
+    const limit = Number(size) || 25;
+    let nextPageUrl = null;
+
+    if (userId) {
+      const applications = await ApplicationModel.getUserApplications(userId);
+
+      return res.json({
+        message: "User applications returned successfully!",
+        applications,
+      });
+    }
+
+    if (status) {
+      const { applications, lastDocId } = await ApplicationModel.getApplicationsBasedOnStatus(status, limit, next);
+
+      if (applications.length === limit) {
+        nextPageUrl = `/applications?next=${lastDocId}&size=${limit}&status=${status}`;
+      }
+
+      return res.json({
+        message: API_RESPONSE_MESSAGES.APPLICATION_RETURN_SUCCESS,
+        applications,
+        next: nextPageUrl,
+      });
```

Comment:
We should try to follow the same as /users and /tasks

Reply:
https://github.com/Real-Dev-Squad/website-backend/issues/1795
as discussed created another ticket for it